infrastructure/src/main/java/org/corfudb/infrastructure/management/ReconfigurationEventHandler.java

```diff
@@ -56,6 +59,28 @@ public static boolean handleFailure(@Nonnull IReconfigurationHandlerPolicy failu
         }
     }
 
+    /**
+     * Calculate a reasonable workflow timeout
+     *
+     * @param runtime Connected corfu runtime instance
+     * @return A reasonable timeout to complete state transfer and rebuild logging unit.
+     */
+    private static Duration getReasonableWorkflowTimeout(CorfuRuntime runtime) {
```

Comment:
The name says Reasonable but that is a tall claim please use a better name. Also this is very static and does not learn from the last failure. Every iteration should learn from the previous failed iteration.